### ISSUE TITLE
python modules _soundcloud_ and _fuzzywuzzy_ were missing

### ISSUE THREAD

**Issue by inecho** [tizonia/tizonia-openmax-il#469]
### Description
tizonia is not able to play music from soundcloud due to missing python modules

### Steps to Reproduce

1. `sudo apt install tizonia-player`
2. set the OAuth token for soundcloud
3. try to play some music by usind `tizonia --soundcloud-creator "creatorsName"`

**Expected behavior:** play some stuff of the creator

**Actual behaviour:** 
```
$ tizonia --soundcloud-creator "rstation"
tizonia 0.13.0. Copyright (C) 2018 Juan A. Rubio
This software is part of the Tizonia project <http://tizonia.org>

[SoundCloud] [Connecting] : 'X-XXXXX-XXXXX-XXXXXXXXXXX'.
Traceback (most recent call last):
  File "<string>", line 2, in <module>
ImportError: No module named fuzzywuzzy

Python modules 'soundcloud' or 'fuzzywuzzy' not found.
Please make sure this is installed correctly.

tizonia exiting (OMX_ErrorInsufficientResources).

 [OMX.Aratelia.audio_source.http:port:0]``
 [OMX_ErrorInsufficientResources]
```
after updating the modules in python over pip, everytime i play some music i get the following warinings:
```
/usr/local/lib/python2.7/dist-packages/requests/__init__.py:80: RequestsDependencyWarning: urllib3 (1.22) or chardet (2.3.0) doesn't match a supported version!
  RequestsDependencyWarning)
/usr/local/lib/python2.7/dist-packages/fuzzywuzzy/fuzz.py:35: UserWarning: Using slow pure-python SequenceMatcher. Install python-Levenshtein to remove this warning
  warnings.warn('Using slow pure-python SequenceMatcher. Install python-Levenshtein to remove this warning')
```
after installing python-Levenshtein, this warning remains:
```
/usr/local/lib/python2.7/dist-packages/requests/__init__.py:80: RequestsDependencyWarning: urllib3 (1.22) or chardet (2.3.0) doesn't match a supported version!
  RequestsDependencyWarning)
```
**Reproduces how often:** [What percentage of the time does it reproduce?]
all the time

### Versions

```
$ tizonia --version 
tizonia 0.13.0. Copyright (C) 2018 Juan A. Rubio
This software is part of the Tizonia project <http://tizonia.org>
-------
$ tizonia --debug
tizonia 0.13.0. Copyright (C) 2018 Juan A. Rubio
This software is part of the Tizonia project <http://tizonia.org>

Debug Info:
	    * [Boost 1_58]
	    * [TagLib 1.9.1]
	    * [MediaInfoLib - v0.7.82]
```

### Installation Method Used

`sudo apt install tizonia-player`

### Additional Information
since i didn't knew for which version of python the code was written, i first updated `pip`, `pip2` and `pip3`. Then installed the modules `soundcloud` and `fuzzywuzzy`

**Comment by tizonia** [comment]
Hi @inecho 

This condition should go away if you use this command on your system (see the Upgrade section on README.md):

```
sudo -H pip install --upgrade gmusicapi soundcloud youtube-dl pafy pycountry titlecase pychromecast plexapi fuzzywuzzy eventlet
```
This assumes pip2 and Python2 are your defaults (otherwise, replace pip with pip2)..

The explanation for this is that Tizonia is a hybrid application that is written in C/C++ and Python. Many of the Python dependencies do not exist as Debian packages, that is why you need to install them separately with pip or other means before or after you install Tizonia's debian packages.

So in general, the 'install.sh' script (see README.md) is the recommended way of installing Tizonia, if you want deb binaries. 'install.sh' will install the Python deps from pips for you. An alternative to that is to use the Snap package, that contains everything if you don't need to worry about the dependencies.

**Comment by tizonia** [comment]
With regards to python-Levenshtein, yes this is an optional dependency. I decided to leave the warning to let people decide if they want to install this additional package or not.

**Comment by inecho** [comment]
Thanks a lot for the quick reply. I used your command above and each package was up to date, but my pip version which was 8.1.1 instead of 9.0.2. Furthermore i was not able to upgrade my pip using the standard approach over `sudo pip install --upgrade pip` due to some inconsistency issues of the installation with the package manager and python itself.. in the end [this one](https://stackoverflow.com/a/46170926/6800468) helped me out. `sudo easy_install -U pip` was the solution. Now everything works fine.  

Btw. is there an overview of keyboard shortcuts? I found out that N is next, P is previous etc.. but i also would like to skim through tracks. is that possible? also: listing playlists over soundcloud is not possible, at least there is no example for that. Should i open a couple more issues according to this?

**Comment by tizonia** [comment]
Regarding pip, yes that's correct. Ubuntu 16.04 shipped with 8.1.1, so it is old at this point. To upgrade, I normally use this method, which also works fine:
https://github.com/tizonia/tizonia-openmax-il/blob/685e5b455546a74044846a86e2e044d58f5e5071/tools/tizonia-dpkg-build#L423

Regarding the shortcuts, there is a little help option that you can use:
![tizonia-help-keyboard](https://user-images.githubusercontent.com/3161606/37724912-b4418cec-2d29-11e8-8842-0c1b0784ecc8.PNG)

As you can see, there is no much there. There is an issue to add seek support (#438). Regarding listing of playlists, I'm not sure what you are referring to. There is an option for playlist search in SoundCloud:

```
$ tizonia --soundcloud-playlists arg
```

If multiple playlists are there with the same name, the first one will be selected and its items added to the playback queue. At the moment, there is no way to modify this behavior via external configuration. The playlist that gets selected, will have its tracks listed on the console before playback of the first track starts. After this point, there is no functionality at the moment to recall the playback queue and list its contents again.

If you are looking for any of these features, feel free to create an issue and add the details there. Suggestions for new features are always welcome. I try to prioritize things according to what I think it will have a greater impact on the project, but I'm also more than happy to accept pull requests if people want to contribute.

Thanks!

**Comment by tizonia** [comment]
Closing this issue as this was resolved by installing the missing python modules.
